### PR TITLE
Add additional entries to troubleshooting section

### PR DESCRIPTION
**Pull request by kevin-bates**
After responding to various issues, I thought it might be good to update the troubleshooting section with additional scenarios and update existing areas that might help save time.

### CODE REVIEW

File: docs/source/troubleshooting.rst [hunk 1]
```diff
@@ -65,22 +65,165 @@ try deleting or renaming that kernelspec folder to expose the default.
...
 you may need to look for support about that kernel.
 
+Python Environments
+-------------------
+Multiple python environments, whether based on Anaconda or Python Virtual environments,
+are often the source of reported issues.  In many cases, these issues stem from the
+Notebook server running in one environment, while the kernel and/or its resources,
+derive from another environment.  Indicators of this scenario include:
+
+* ``import`` statements within code cells producing ``ModuleNotFound`` exceptions.
```
Comment (Zsailer) [inline docs/source/troubleshooting.rst:75]: I think Python <3.5 raises an `ImportError`.

File: docs/source/troubleshooting.rst [hunk 2]
```diff
@@ -65,22 +65,165 @@ try deleting or renaming that kernelspec folder to expose the default.
...
+* General kernel startup failures exhibited by nothing happening when attempting
+  to execute a cell.
+
+In these situations, take a close look at your environment structure and ensure all
+packages required by your notebook's code are installed in the correct environment.
+If you need to run the kernel from different conda environments than your Notebook
```
Comment (Zsailer) [inline docs/source/troubleshooting.rst:81]: I think we should reference [these docs](https://ipython.readthedocs.io/en/stable/install/kernel_install.html#kernels-for-different-environments) **before** `nb_conda_kernels`. It describes the Jupyter supported way for installing kernels for other conda environments.

Comment (kevin-bates) [reply docs/source/troubleshooting.rst]: Thanks Zach - I was not aware of that.  An update has been added.

File: docs/source/troubleshooting.rst [hunk 3]
```diff
@@ -65,22 +65,168 @@ try deleting or renaming that kernelspec folder to expose the default.
...
 you may need to look for support about that kernel.
 
+Python Environments
+-------------------
+Multiple python environments, whether based on Anaconda or Python Virtual environments,
+are often the source of reported issues.  In many cases, these issues stem from the
+Notebook server running in one environment, while the kernel and/or its resources,
+derive from another environment.  Indicators of this scenario include:
+
+* ``import`` statements within code cells producing ``ModuleNotFound`` exceptions.
```
Comment (Zsailer) [inline docs/source/troubleshooting.rst:75]:
```suggestion
* ``import`` statements within code cells producing ``ImportError`` or ``ModuleNotFound`` exceptions.
```

**Comment by Zsailer** [comment]
This is awesome, @kevin-bates! Thank you for taking the time to flesh out this documentation!

**Pull request merged**